tools/codegen/pkg/interfacegen/generator_test.go

```diff
@@ -15,159 +15,89 @@
-				cmd.Dir = outDir
-				cmd.Stderr = os.Stderr // For debugging
-				err = cmd.Run()
-				if err != nil {
-					t.FailNow()
-					return
-				}
-			} else {
-				_ = ioutil.WriteFile(outFilePath, []byte(err.Error()), 0644)
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			if same := deepCompare(v.got, v.want, t.Errorf); !same {
+				t.Error("Files were not the same.")
 			}
-
-			diffCmd := exec.Command("diff", outFilePath, tt.baseline, "--ignore-all-space")
-			diffCmd.Stdout = os.Stdout
-			diffCmd.Stderr = os.Stderr
-			err = diffCmd.Run()
-			if err != nil {
-				t.Fatalf("Diff failed: %+v. Expected output is located at %s", err, outFilePath)
-				return
-			}
-
-			// if the test succeeded, clean up
-			_ = os.RemoveAll(outDir)
 		})
 	}
 }
 
-func removeDir(dir string) {
-	_ = os.Remove(dir)
-}
-
-// TODO: This is blocking the test to be enabled from Bazel.
-func generteFDSFileHacky(protoFile string, outputFDSFile string) error {
-
-	// HACK HACK. Depending on dir structure is super fragile.
-	// Explore how to generate File Descriptor set in a better way.
-	protocCmd := []string{
-		path.Join("mixer/tools/codegen/pkg/interfacegen", protoFile),
-		"-o",
-		path.Join("mixer/tools/codegen/pkg/interfacegen", outputFDSFile),
-		"-I=.",
-		"-I=api",
-		"--include_imports",
-		"--include_source_info",
+func TestGenerate_Errors(t *testing.T) {
+	g := Generator{OutFilePath: "."}
+	err := g.Generate("testdata/error_template.descriptor_set")
+	if err == nil {
+		t.Fatalf("Generate(%s) should have produced an error", "testdata/error_template.descriptor_set")
 	}
-	cmd := exec.Command("protoc", protocCmd...)
-	dir := path.Join(os.Getenv("GOPATH"), "src/istio.io")
-	cmd.Dir = dir
-	cmd.Stderr = os.Stderr // For debugging
-	err := cmd.Run()
-	return err
-}
-
-// TODO: This is blocking the test to be enabled from Bazel.
-func generteGoPbFileForTmpl(protoFile string, outDir string) error {
-
-	// HACK HACK. Depending on dir structure is super fragile.
-	// Explore how to generate File Descriptor set in a better way.
-	protocCmd := []string{
-		path.Join("mixer/tools/codegen/pkg/interfacegen", protoFile),
-		"--go_out=Mgoogle/protobuf/duration.proto=github.com/golang/protobuf/ptypes/duration,Mmixer/v1/" +
-			"config/descriptor/value_type.proto=istio.io/api/mixer/v1/config/descriptor," +
-			"Mgoogle/protobuf/descriptor.proto=github.com/golang/protobuf/protoc-gen-go/descriptor," +
-			"Mgoogle/protobuf/struct.proto=github.com/golang/protobuf/ptypes/struct," +
-			"Mmixer/tools/codegen/pkg/template_extension/TemplateExtensions.proto=istio.io/mixer/tools/codegen/pkg/template_extension:.",
-		"-I=.",
-		"-I=api",
+	b, fileErr := ioutil.ReadFile("testdata/ErrorTemplate.baseline")
+	if fileErr != nil {
+		t.Fatalf("Could not read baseline file: %v", err)
 	}
+	want := fmt.Sprintf("%s", b)
+	got := err.Error()
+	if got != want {
+		t.Fatalf("Generate(%s) => '%s'\nwanted: '%s'", "testdata/error_template.descriptor_set", got, want)
```

Comment:
Can we print the file where the got file will be placed. Will be easy to go and copy/replace when creating new tests  ?